api/v1alpha1/paasconfig_types.go

```diff
@@ -60,12 +60,12 @@ type PaasConfigSpec struct {
 
 	// A map with zero or more ConfigCapability
 	// +kubebuilder:validation:Required
```

Comment:
This is perfectly valid for controller runtime, however looks ugly as we have two conflicting instructions. Please use one of the following:

a. use `// +kubebuilder:validation:Optional` instead of adding `omitempty`
b. remove kubebuilder marker and use `omitemty`
c. declare explicitly both Optional and `omitempty`

personal preference is: a.

Reply:
Fair point.
I have replaced all `omitempty` flags with `// +kubebuilder:validation:Optional` (replacing `Required` with `Optional` wherever needed.
We could do the same for paas_types.go and paasns_types.go, but I left them unchanged for now.